tests/python_functional/functional_tests/conftest.py

```diff
@@ -74,7 +74,11 @@ class PortAllocator():
     CURRENT_DYNAMIC_PORT = 30000
 
 
-@pytest.fixture(scope="module")
-def some_port():
+def _get_next_port():
```

Comment:
if your intention was to mark _get_next_port() as a private function, pls use a double underscore in the name.
__get_next_port

Reply:
If you want this function to be private, then instead of adding underscores,  you could make it a local function inside the `port_allocator`:

```python
@pytest.fixture(scope="session")
def port_allocator():
    def get_next_port():
        PortAllocator.CURRENT_DYNAMIC_PORT += 1
        return PortAllocator.CURRENT_DYNAMIC_PORT

    return get_next_port
```
(note that I did not underscore inside the `port_allocator` either)

What is the reason behind  `_get_next_port` to be private?

In general, I do not really like the effort we are putting into making things private by adding underscores. We are doing this way more often than it is reasonable. One downside is that it is harder to write unit test for such code. I do not find start with underscores aesthetically pleasing either. Sometimes, they can also get in the way: when I really want to call these functions, it is frustrating to sed the names. This would worth a discussion with the whole team. I thnk there are better ways for achieve information hiding.